File: Antony-Ndungu/Arrays and Strings 1/minimum-swaps-2.go
package main

import "fmt"

func main() {
	nums := []int{2, 3, 4, 1, 5}
	fmt.Println(minimumSwaps(nums))
}

func minimumSwaps(nums []int) int {
	var numSwaps, index int
	for index < len(nums) {
		if (index + 1) != nums[index] {
			nums[index], nums[nums[index]-1] = nums[nums[index]-1], nums[index]
			numSwaps++
		} else {
			index++
		}
	}
	return numSwaps
}

File: Antony-Ndungu/Arrays and Strings 1/minimum-swaps-2.c
#include <stdio.h>

int minimumSwaps(int *arr, size_t arrSize);

int main(void){
    int nums[] = {2, 3, 4, 1, 5};
    size_t numsSize = sizeof(nums) / sizeof(int);
    printf("%d\n", minimumSwaps(nums, numsSize));
    return 0;
}

int minimumSwaps(int *arr, size_t arrSize){
    int numSwaps = 0, index = 0, swapIndex, temp;
    while(index < arrSize){
        if(index + 1 != arr[index]){
            swapIndex = arr[index] - 1;
            temp = arr[swapIndex];
            arr[swapIndex] = arr[index];
            arr[index] = temp;
            numSwaps++;
        }else{
            index++;
        }
    }
    return numSwaps;
}